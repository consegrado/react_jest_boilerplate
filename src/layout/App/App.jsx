// @flow

import React from 'react';
import TabsDemo from '@/components/TabsDemo';

const App = () => {
  return (
    <>
      <TabsDemo />
    </>
  );
};

export default App;
