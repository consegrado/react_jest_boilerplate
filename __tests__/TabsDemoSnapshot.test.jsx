// @flow

import React from 'react';
import App from '@/layout/App';
import { mount } from 'enzyme';

describe('App', () => {
  it('render', () => {
    const wrapper = mount(<App />);
    expect(wrapper.render()).toMatchSnapshot();
    const tabLast = wrapper.find('[data-test="tab"]').last();
    tabLast.simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });
});
