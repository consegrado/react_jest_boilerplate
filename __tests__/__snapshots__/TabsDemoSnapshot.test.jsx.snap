// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render 1`] = `
<div
  data-test="demo-tabs"
>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        <span
          data-test="tab"
        >
          Title 0
        </span>
        <button
          data-test="tab-remove"
          type="button"
        >
          remove
        </button>
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        id="react-tabs-2"
        role="tab"
      >
        <span
          data-test="tab"
        >
          Title 1
        </span>
        <button
          data-test="tab-remove"
          type="button"
        >
          remove
        </button>
      </li>
      <button
        data-test="tab-add"
        type="button"
      >
        add
      </button>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      data-test="tab-content"
      id="react-tabs-1"
      role="tabpanel"
    >
      <div>
        <h2>
          Any content 0
        </h2>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      data-test="tab-content"
      id="react-tabs-3"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`App render 2`] = `
<div
  data-test="demo-tabs"
>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        id="react-tabs-0"
        role="tab"
      >
        <span
          data-test="tab"
        >
          Title 0
        </span>
        <button
          data-test="tab-remove"
          type="button"
        >
          remove
        </button>
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-2"
        role="tab"
        tabindex="0"
      >
        <span
          data-test="tab"
        >
          Title 1
        </span>
        <button
          data-test="tab-remove"
          type="button"
        >
          remove
        </button>
      </li>
      <button
        data-test="tab-add"
        type="button"
      >
        add
      </button>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel"
      data-test="tab-content"
      id="react-tabs-1"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      data-test="tab-content"
      id="react-tabs-3"
      role="tabpanel"
    >
      <div>
        <h2>
          Any content 1
        </h2>
      </div>
    </div>
  </div>
</div>
`;
